Framework/PythonInterface/mantid/api/src/Exports/AlgorithmProperty.cpp

```diff
@@ -39,9 +39,10 @@ AlgorithmProperty *createPropertyWithValidatorAndDirection(const std::string &na
  * when passed to the framework
  * @return A pointer to a new AlgorithmProperty object
  */
-AlgorithmProperty *createPropertyWithValidator(const std::string &name, IValidator *validator) {
+static const AlgorithmProperty *(*const createPropertyWithValidator)(const std::string &, const IValidator *) =
+    +[](const std::string &name, const IValidator *validator) -> const AlgorithmProperty * {
```

Comment:
Can you explain this syntax?

Reply:
I think in this case the code has become quite difficult to read, so maybe this one can be omitted from this PR.

Reply:
Do you mean finding a better solution for the suppression or resolving it in another PR?

Reply:
I think put the suppression back into the suppressions file for now so that we can merge this PR and have a look at it separately.

Reply:
OK, I will revert the change